modules/http-helper/dummy_server.go

```diff
@@ -44,6 +44,39 @@ func RunDummyServerE(t *testing.T, text string) (net.Listener, int, error) {
 	return listener, port, err
 }
 
+// RunDummyHandlerServer runs a dummy HTTP server on a unique port that will serve given handlers. Returns the Listener for the server,
+// the port it's listening on, or an error if something went wrong while trying to start the listener. Make sure to call
+// the Close() method on the Listener when you're done!
+func RunDummyHandlerServer(t *testing.T, handlers map[string]func(http.ResponseWriter, *http.Request)) (net.Listener, int) {
+	listener, port, err := RunDummyHandlerServerE(t, handlers)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return listener, port
+}
+
+// RunDummyHandlerServerE runs a dummy HTTP server on a unique port that will server given handlers. Returns the Listener for the server,
+// the port it's listening on, or an error if something went wrong while trying to start the listener. Make sure to call
+// the Close() method on the Listener when you're done!
+func RunDummyHandlerServerE(t *testing.T, handlers map[string]func(http.ResponseWriter, *http.Request)) (net.Listener, int, error) {
+	port := getNextPort()
+
+	for path, handler := range handlers {
+		http.HandleFunc(path, handler)
```

Comment:
Isn't this a global call? If so, then using this across multiple tests in parallel will cause conflicts, right?

IIRC, there was some way with Go's HTTP server to define a local instance of the server and its handler functions.